modules/social_features/social_core/src/Service/ConfigTranslator.php

```diff
@@ -0,0 +1,105 @@
+ *
+ * Usage:
+ * 1. open_config_translator($langcode)
+ * 2. Load configuration in correct language
+ * 3. close_config_translator()
+ *
+ * Explanation on example why this is needed:
+ *
+ *    Accessing translated configuration
+ *
+ *    Drupal by default always uses the language selected for the page to load
+ *    configuration with. So if you are viewing a Spanish page, all
+ *    configuration is loaded with Spanish translations merged in. If you are
+ *    viewing the same page in Hungarian, the same code will now receive
+ *    Hungarian translated configuration. This means normally you don't need to
+ *    do anything special to access configuration in the language needed.
+ *
+ *    However, there are cases, when you want to load the original copy of the
+ *    configuration or ask for a specific language. Such as when sending emails
+ *    to users, you will need configuration values in the right language.
+ *    The following code is a slightly adapted excerpt from user_mail() to
+ *    illustrate loading configuration in the preferred language of $account to
+ *    compose an email:
+ *
+ *    <?php
+ *      $language_manager = \Drupal::languageManager();
+ *      $language = $language_manager->getLanguage($account->getPreferredLangcode());
+ *      $original_language = $language_manager->getConfigOverrideLanguage();
+ *      $language_manager->setConfigOverrideLanguage($language);
+ *      $mail_config = \Drupal::config('user.mail');
+ *        // ...
+ *        // ...Compose (and send) email here...
+ *        // ...
+ *      $language_manager->setConfigOverrideLanguage($original_language);
+ *    ?>
+ *
+ *    Note that you are setting values on the language manager and not the
+ *    configuration system directly. The configuration system is
+ *    override-agnostic and can support overrides of different kinds with
+ *    various conditions. It is the job of the overrides to manage their
+ *    conditions, in this case to allow changing the language used.
+ *    The same pattern can be used to load configuration entities in specific
+ *    languages as well.
+ *
+ *    Source: https://www.hojtsy.hu/blog/2014-may-26/drupal-8-multilingual-tidbits-16-configuration-translation-development
+ */
+class ConfigTranslator extends ControllerBase {
+
+  /** @var Language */
+  public $translation_language;
+
+  /** @var Language */
+  public $original_language;
+
+  /**
+   * @inerhitDoc
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('language_manager'),
+    );
+  }
+
+  /**
+   * ConfigTranslator constructor.
+   * @param LanguageManager $languageManager
+   */
+  public function __construct(LanguageManager $languageManager) {
+    $this->languageManager = $languageManager;
+  }
+
+  /**
+   * Define language in which configuration should be processed.
+   *
+   * @param string $langcode
+   */
+  public function open_config_translator(string $langcode): void {
```

Comment:
Please, according to drupal coding standards functions inside classes should be declared with camelCaps: https://www.drupal.org/docs/develop/standards/coding-standards#s-parameter-and-return-typehinting